docs/reference/snapshot-restore/repository-s3.asciidoc

```diff
@@ -292,6 +293,14 @@ include::repository-shared-settings.asciidoc[]
     size allowed by S3. Defaults to `100mb` or `5%` of JVM heap, whichever is
     smaller.
 
+`max_multipart_parts` ::
+
+    (<<number,numeric>>) Maximum number of parts for multipart upload. When large file split into
+    chunks, the chunk size will be defined by smallest of `chunk_size` or `buffer_size * max_multipart_parts`.
+    Default value is 10,000, also see https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html[S3 multipart upload limits].
+    For example, with `buffer_size=100MB` and `max_multipart_parts=10,000` summation of all parts is about 1TB.
+    If chunk_size is set to 5TB then smallest between two would be 1TB.
```

Comment:
I'd be inclined not to dwell too much on the calculations here, how about something like this?

```suggestion
    (<<number,integer>>) The maximum number of parts that {es} will write during a multipart upload of a single object. Files which are
    larger than `buffer_size × max_multipart_parts` will be chunked into several smaller objects. {es} may also split a file across multiple
    objects to satisfy other constraints such as the `chunk_size` limit. Defaults to `10000` which is the
    https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html[maximum number of parts in a multipart upload in AWS S3].
```